### ISSUE TITLE
Refactor Categories.getCategoryById

### ISSUE THREAD

**Issue by julianlam** [NodeBB/NodeBB#962]
[These lines](https://github.com/designcreateplay/NodeBB/blob/d2bed7791c4009d9a3ca08b3693b7105423c69cc/src/categories.js#L55-L71) define functions that will be called in parallel to load prerequisite data when retrieving category data.

[These lines](https://github.com/designcreateplay/NodeBB/blob/d2bed7791c4009d9a3ca08b3693b7105423c69cc/src/categories.js#L100-L106) retrieve moderator and active user lists.
### Issues
- There is no need for these to wait for [`getCategoryData`](https://github.com/designcreateplay/NodeBB/blob/d2bed7791c4009d9a3ca08b3693b7105423c69cc/src/categories.js#L50) to return, since all they require is a `cid`. `getCategoryData` should be part of the parallel execution
- There is no reason why the moderators list needs to wait for the category data to be compiled (as it just required `cid` again) -- it should be part of the first parallel execution block.
- `getActiveUsers` (line 104), curiously, is _re-defining_ the earlier defined `getActiveUsers`. In the earlier function, it retrieves cids. In the later function, it retrieves user data. They can be merged into one (or the call to `getMultipleUserFields` can be merged into the parallel execution block).
